cloud_images/centos7/install_prereqs.sh

```diff
@@ -35,7 +37,7 @@ ExecStart=/usr/local/sbin/dcos_vol_setup.sh /dev/xvdg /dcos/volume0
 ExecStart=/usr/local/sbin/dcos_vol_setup.sh /dev/xvdh /var/log
 
 [Install]
-WantedBy=local-fs.target
+RequiredBy=local-fs-pre.target
```

Comment:
I know you're asking @lloesche, but I feel the need to emphasize that this change is one of the three critical changed lines in this PR (the others are the DefaultDependencies=no, and sending journald a sighup instead of restarting it).

1. Before this PR, the systemd unit was attempting to pull in local-fs.target with `WantedBy`. Coincidentally that was not working correctly because the default implicit dependencies were not disabled (see attached systemd analyze graphs). We could have left this with WantedBy, but we probably don't want the formatting to happen if local-fs.target fails for any reason, so RequiredBy seemed to make more sense.
2. local-fs-pre.target is an earlier hook than local-fs.target in the systemd sequence. Also, local-fs.target is intended for fsck'ing volumes, so we really do want to run before local-fs (https://www.freedesktop.org/software/systemd/man/bootup.html). And  since we are formatting partitions that are going to be used by other services, it is best to do this as early as possible.